tools/cargo-to-recipe.sh

```diff
@@ -0,0 +1,287 @@
+#!/usr/bin/env bash
+
+die() {
+	printf '%s' "${*+$'\e[31mError: \e[0m'$*$'\n'}"
+	(( usage )) && usage
+	exit 1
+} 1>&2
+
+usage() {
+	cat <<- EOF
+	Usage: $0 [options] URI category/port
+
+	Creates a recipe template for a crates.io package, filled with
+	information at hand.
+
+	Options:
+	  -h, --help	show this help message and exit
+	  -k, --keep	keep the generated temporary directory
+	  -psd, --print-source-directories
+	 		also print SOURCE_DIRs
+	  -c CMD, --cmd=CMD
+	 		specify the command runtime
+	  -b PORTNAME, --bump PORTNAME
+	 		bump the crates.io dependencies of the specified port
+	EOF
+}
+
+temp() { rm -rf "$tempdir"; }
+
+. "$(finddir B_USER_SETTINGS_DIRECTORY)"/haikuports.conf
+while (( $# )); do
+	case "$1" in
+		-h|--help)
+			usage
+			exit 0
+			;;
+		-k|--keep)
+			temp() { printf '%s\n' "Kept $tempdir"; }
+			;;
+		-psd|--print-source-directories)
+			psd=3
+			;;
+		-c)
+			shift
+			;&
+		--cmd=*)
+			cmd=${1#*=}
+			;;
+		-b|--bump)
+			portName=$2
+			directory=$(
+				find "$TREE_PATH" -mindepth 3 -maxdepth 3 \
+					-iname "$portName-*.*.recipe" |
+					awk 'FNR == 1 { gsub("/[^/]*$", "")
+							print }'
+			)
+			bump=1
+			shift
+			;;
+		*://*)
+			SOURCE_URI=$1
+			;;
+		*-*/*)
+			directory=$TREE_PATH/$1
+			: "${1//-/_}"
+			portName=${_#*/}
+			;;
+		-?*)
+			usage=1 die "Invalid option."
+			;;
+		*)
+			usage=1 die "Invalid argument."
+			;;
+	esac
+	shift
+	false
+done && usage=1 die
+
+mkdir -p "$directory"/download
+cd "$directory" || die "Invalid port directory."
+
+if (( bump )); then
+	eval "recipe=$(
+		ls -v --quoting-style=shell-escape \
+			"$portName"-[0-9]*.[0-9]*.recipe | tail -n 1
```

Comment:
I think `find` should be used here instead, and why was this in an `eval`?

Reply:
`eval` is used with `--quoting-style=shell-escape` to correctly parse `ls` (see 
https://mywiki.wooledge.org/ParsingLs); it probably isn't necessary here as recipes shouldn't have any tricky characters, but it's good practice nonetheless. I used `ls` for the version sort, although I suppose that I could pipe `find` to `sort --version-sort` instead.